pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -428,10 +437,19 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 	}
 
 	retry := backoff.NewRetry(retryPushNum, backoff.NewConstant(retryPushInterval))
-	_, err = retry.Do(ctx, func() (interface{}, error) {
-		err := tmpRepo.Push(ctx, tmpRepo.GetClonedBranch())
-		return nil, err
-	})
+	for _, branch := range newBranchs {
+		_, err = retry.Do(ctx, func() (interface{}, error) {
+			err := tmpRepo.Push(ctx, branch)
+			return nil, err
+		})
+	}
+	if isExistDefaultBranch {
+		_, err = retry.Do(ctx, func() (interface{}, error) {
+			err := tmpRepo.Push(ctx, repo.GetClonedBranch())
+			return nil, err
+		})
+	}
```

Comment:
Thank you!
I fixed in this ! [use map for branchname](https://github.com/pipe-cd/pipecd/pull/4395/commits/c6ae27ad654307aad3548b4837a60f2af6e2ab8b)
I used `branchNames` map instead of `newBranches`, because the map has new branch name and default branch name.